src/ddmd/mars.d

```diff
@@ -1506,6 +1529,11 @@ Language changes listed by -transition=id:
     // So deps file generation should be moved after the inlinig stage.
     if (global.params.moduleDeps)
     {
+        StringTable importArray;
```

Comment:
Nice! Thanks. I moved the function to dmodule.d and replaced the AA with a check to semanticRun.